kernel/src/process.rs

```diff
@@ -933,11 +933,32 @@ impl<'a> Process<'a> {
             if let Some(load_result) =
                 load(tbf_header, remaining_app_memory) {
 
+                // First determine how much space we need in the application's
+                // memory space just for kernel and grant state. We need to make
+                // sure we allocate enough memory just for that.
+
+                // Make room for grant pointers.
+                let grant_ptr_size = mem::size_of::<*const usize>();
+                let grant_ptrs_num = read_volatile(&grant::CONTAINER_COUNTER);
+                let grant_ptrs_offset = grant_ptrs_num * grant_ptr_size;
+
+                // Allocate memory for callback ring buffer.
+                let callback_size = mem::size_of::<Task>();
+                let callback_len = 10;
```

Comment:
Where does this magic 10 come from?

Reply:
I don't know, it was already there.

Reply:
That's from the early, early days of Tock. We put a simple, fixed-length queue in for callbacks because it was easy at the time and I don't think we settled on the more general solution for nesting/repeated callbacks. It's also the reason that https://github.com/helena-project/tock/issues/536 drops the 11th, though the underlying issue there is orthogonal to the size of this number.

Reply:
It makes sense, we have a fixed-size queue. But it seems problematic that this is a magic constant re-defined in multiple files, rather than being defined in one place and used in many?

Reply:
Where else is it defined? I agree it being magic isn't awesome (I think at some point we wanted it to at least be process tunable, which we can still do), but I _think_ it's only defined here.